### PR TITLE
Use a YAML file to store product information

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request is focused on switching to using a YAML file to store information about affected products. This is achieved through the following:

- Use the existing `SOFTWARE-LIST.md` contents to populate an appropriately formatted YAML file.
- Add a GitHub Actions workflow that will update the contents of `SOFTWARE-LIST.md` when a `push` event occurs on the `develop` branch. This will use the current contents of `data/cisagov.yml` (the YAML file) to populate `SOFTWARE-LIST.md`.

## 💭 Motivation and context ##

Using a machine readable file with stricter formatting requirements to store this information should reduce human error and also make it easier for third parties to ingest our data.

## 🧪 Testing ##

Automated tests pass. I tested this workflow configuration in [this branch of a testing repository](https://github.com/cisagov/github-actions-playground/tree/testing/log4j_list_update) and it worked as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by justmurphy — APPROVED** [review]
Looks great!

File: .github/workflows/update_software_list.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          check-name: lint
+          ref: ${{ needs.setup.outputs.testing_branch }}
+          repo-token: ${{ github.token }}
+      - name: Merge the testing branch
+        run: |
+          git fetch
+          git merge origin/${{ needs.setup.outputs.testing_branch }}
+          git push
+      - name: Cleanup testing branch
+        run: git push -d origin ${{ needs.setup.outputs.testing_branch }}
```
Comment (dav3r) [inline .github/workflows/update_software_list.yml]: Prefer full name of switch:
```suggestion
        run: git push --delete origin ${{ needs.setup.outputs.testing_branch }}
```

**Review by dav3r — APPROVED** [review]
This looks fantastic to me - excellent work as always! 💪 

I mentioned one tiny thing to change, but otherwise it's great.

**Review by iainDe — APPROVED** [review]

File: .github/workflows/update_software_list.yml (hunk 1 above)
Comment (mcdonnnj) [reply .github/workflows/update_software_list.yml]: Good catch!

**Pull request merged**